### DISCUSSION TITLE
Raspberry Pi 3 + RPi OS Bookworm

### DISCUSSION THREAD

**Discussion by SvenVD** [OpenSurv/OpenSurv#4] [Hardware + Operating system]
All experiences and tests with OpenSurv on Raspberry Pi 3 
( old thread https://github.com/SvenVD/rpisurv/issues/195)

For Ubuntu 24.04 LTS see https://github.com/OpenSurv/OpenSurv/discussions/20

**Comment by meetyourlaser** [comment]
Works on Raspberry Pi 3 (Model B Plus Rev 1.3) with Raspberry Pi OS (Debian GNU/Linux 12 (Bookworm)).

I had to install _python3-yaml_ (`apt install python3-yaml`) in addition to following the OpenSurv installation instructions.

I have only tested it with one monitor (1920x1080). It works well with one stream at a time (total of three streams).

P.S.: The same setup also works on a Raspberry Pi 2 (Model B Rev 1.1) with Raspberry Pi OS.

**Reply by SvenVD** [reply]
Thanks for reporting back

**Reply by UntouchedWagons** [reply]
I tried using the same hardware and OS and opensurv never starts, I just get the desktop manager login screen.

**Reply by builder11a** [reply]
> I tried using the same hardware and OS and opensurv never starts, I just get the desktop manager login screen.

I'm having this same issue. @SvenVD, can you help with this?  

2025/03/14 17:51:58 - DAEMON:
2025/03/14 17:51:58 - DAEMON: (xfwm4:664): GLib-CRITICAL **: 17:51:58.869: g_str_has_prefix: assertion 'prefix != NULL' failed
2025/03/14 17:51:59 - DAEMON:
2025/03/14 17:51:59 - DAEMON: (xfwm4:664): xfwm4-WARNING **: 17:51:59.000: The property '/general/double_click_distance' of type int is not supported
2025/03/14 17:51:59 - DAEMON: Traceback (most recent call last):
2025/03/14 17:51:59 - DAEMON: File "surveillance.py", line 7, in <module>
2025/03/14 17:51:59 - DAEMON: from core.util.config import cfg
2025/03/14 17:51:59 - DAEMON: File "/home/opensurv/lib/core/util/config.py", line 1, in <module>
2025/03/14 17:51:59 - DAEMON: import yaml
2025/03/14 17:51:59 - DAEMON: ModuleNotFoundError: No module named 'yaml'
2025/03/14 17:51:59 - DAEMON:
2025/03/14 17:51:59 - DAEMON: (xfwm4:664): xfwm4-WARNING **: 17:51:59.171: Error opening /dev/dri/card0: No such file or directory
2025/03/14 17:51:59 - DAEMON: xfwm4: Fatal IO error 0 (Success) on X server :0.
2025/03/14 17:51:59 - DAEMON: XIO:  fatal IO error 2 (No such file or directory) on X server ":0"
2025/03/14 17:51:59 - DAEMON: after 20 requests (20 known processed) with 0 events remaining.

**Reply by SvenVD** [reply]
> > I tried using the same hardware and OS and opensurv never starts, I just get the desktop manager login screen.
> 
> I'm having this same issue. @SvenVD, can you help with this?
> 
> 2025/03/14 17:51:58 - DAEMON: 2025/03/14 17:51:58 - DAEMON: (xfwm4:664): GLib-CRITICAL **: 17:51:58.869: g_str_has_prefix: assertion 'prefix != NULL' failed 2025/03/14 17:51:59 - DAEMON: 2025/03/14 17:51:59 - DAEMON: (xfwm4:664): xfwm4-WARNING **: 17:51:59.000: The property '/general/double_click_distance' of type int is not supported 2025/03/14 17:51:59 - DAEMON: Traceback (most recent call last): 2025/03/14 17:51:59 - DAEMON: File "surveillance.py", line 7, in 2025/03/14 17:51:59 - DAEMON: from core.util.config import cfg 2025/03/14 17:51:59 - DAEMON: File "/home/opensurv/lib/core/util/config.py", line 1, in 2025/03/14 17:51:59 - DAEMON: import yaml 2025/03/14 17:51:59 - DAEMON: ModuleNotFoundError: No module named 'yaml' 2025/03/14 17:51:59 - DAEMON: 2025/03/14 17:51:59 - DAEMON: (xfwm4:664): xfwm4-WARNING **: 17:51:59.171: Error opening /dev/dri/card0: No such file or directory 2025/03/14 17:51:59 - DAEMON: xfwm4: Fatal IO error 0 (Success) on X server :0. 2025/03/14 17:51:59 - DAEMON: XIO: fatal IO error 2 (No such file or directory) on X server ":0" 2025/03/14 17:51:59 - DAEMON: after 20 requests (20 known processed) with 0 events remaining.

Did you do `apt install python3-yaml` like @meetyourlaser suggested? It seems to be missing the yaml module

**Reply by builder11a** [reply]
> > > I tried using the same hardware and OS and opensurv never starts, I just get the desktop manager login screen.
> > 
> > 
> > I'm having this same issue. @SvenVD, can you help with this?
> > 2025/03/14 17:51:58 - DAEMON: 2025/03/14 17:51:58 - DAEMON: (xfwm4:664): GLib-CRITICAL **: 17:51:58.869: g_str_has_prefix: assertion 'prefix != NULL' failed 2025/03/14 17:51:59 - DAEMON: 2025/03/14 17:51:59 - DAEMON: (xfwm4:664): xfwm4-WARNING **: 17:51:59.000: The property '/general/double_click_distance' of type int is not supported 2025/03/14 17:51:59 - DAEMON: Traceback (most recent call last): 2025/03/14 17:51:59 - DAEMON: File "surveillance.py", line 7, in 2025/03/14 17:51:59 - DAEMON: from core.util.config import cfg 2025/03/14 17:51:59 - DAEMON: File "/home/opensurv/lib/core/util/config.py", line 1, in 2025/03/14 17:51:59 - DAEMON: import yaml 2025/03/14 17:51:59 - DAEMON: ModuleNotFoundError: No module named 'yaml' 2025/03/14 17:51:59 - DAEMON: 2025/03/14 17:51:59 - DAEMON: (xfwm4:664): xfwm4-WARNING **: 17:51:59.171: Error opening /dev/dri/card0: No such file or directory 2025/03/14 17:51:59 - DAEMON: xfwm4: Fatal IO error 0 (Success) on X server :0. 2025/03/14 17:51:59 - DAEMON: XIO: fatal IO error 2 (No such file or directory) on X server ":0" 2025/03/14 17:51:59 - DAEMON: after 20 requests (20 known processed) with 0 events remaining.
> 
> Did you do `apt install python3-yaml` like @meetyourlaser suggested? It seems to be missing the yaml module

Yep, the file wasn't included in the OS, so had to get it, and then install.  Looks like same error in the logs after.  OP above must have done something else as well.  @meetyourlaser

**Comment by TBXtbx** [comment]
Hi,
I am testing OpenSurv on a Raspberry Pi 2 Model B Rev 1.1 running Raspberry Pi OS.
I confirm that you need to install python3-yaml.

For the moment everything seems to work OK.
But in the logs I see entries:
V: 00:28:39 / 00:28:40 (100%) Dropped: 4
V: 00:28:46 / 00:28:47 (100%) Dropped: 4
V: 00:30:24 / 00:30:25 (100%) Dropped: 4

What do they mean by the number of lost frames ?

**Reply by SvenVD** [reply]
It probably means they are dropped because hardware could not handle them in time at that time, but 4 is not a big number. Is it increasing fast?

**Comment by TBXtbx** [comment]
Last 24h

V: 43:34:46 / 43:34:47 (100%) Dropped: 341
V: 43:35:04 / 43:35:05 (100%) Dropped: 341
V: 43:35:06 / 43:35:07 (100%) Dropped: 341
V: 43:35:33 / 43:35:34 (100%) Dropped: 341
V: 43:35:37 / 43:35:38 (100%) Dropped: 341
V: 43:35:39 / 43:35:40 (100%) Dropped: 341
V: 43:35:48 / 43:35:49 (100%) Dropped: 341


The bigger problem for me currently is the delay which is about 5 sec. - Rpi 2 B - equipment fails :) 
Rpi 3 - delay 2 sec.
How to reduce the delay ?

**Reply by SvenVD** [reply]
341 dropped in 24h is not a big deal

**Comment by tuupos** [comment]
I also have a stable Raspberry Pi 3 - Model B Plus Rev 1.3 - RPi OS Bookworm.

4 streams. Al tough every day one camera "hangs". then restart of the service gets is back online. I now do a cron job to restart the rpi every night at a given time.

**Comment by TBXtbx** [comment]
I experienced hangs on Rpi3 B rev. 1.2 on Rspisurv - several times a day. Try changing the camera stream settings.
Me thinks that Rpi 3 for 4 streams is too weak.
At my place 4 streams relatively stable work only on Rpi 4

On OpenSurv it works stably on Rpi2 and Rpi3 B rev 1.2. But I am testing 1 stream for now. 
Only I have a delay of 3 sec and I do not know how to deal with it to reduce it a little more.

**Reply by SvenVD** [reply]
I have sub second latency on my N100 chipset when I enable https://github.com/OpenSurv/OpenSurv/blob/e904ccc128fe56990242b801af3437302a5d79d1/README.md?plain=1#L133

**Reply by TBXtbx** [reply]
Bingo, delay 0 sec.
Great job @SvenVD

**Reply by ChRoNe-IT** [reply]
I added this also but my streams freezes between 1-4 minutes after reconnecting or restarting and never get refreshed again

**Comment by SvenVD** [comment]
For your information if someone is using Raspberry Pi 3 with Ubuntu 24.04 LTS:  https://github.com/OpenSurv/OpenSurv/discussions/20. This discussion will be dedicated to bookworm

**Comment by SvenVD** [comment]
https://github.com/OpenSurv/OpenSurv/issues/24#issuecomment-2561011670

**Comment by jcrossbdn** [comment]
running a pi3b+ on bookworm with a single stream (an instance of rpisurv feeding a hardware encoder to send a single rtsp stream to this opensurv instance over wifi).

It will run for a while and then logs show:
2025/01/22 16:01:08 - l_default - ERROR - Stream: screen_manager_0_screen1_stream1: run_stream_watchdog: is not responding trying to restart
2025/01/22 16:01:17 - l_default - ERROR - Stream: screen_manager_0_screen1_stream1: _wait_for_window_to_be_initialized, window not found (14/14. Stopping attempt

**Reply by TBXtbx** [reply]
Hi,
Make sure it's not your WiFi problem.
Check Ripi's operation over LAN RJ45.

**Reply by SvenVD** [reply]
On (very) slow starting streams you can increase the waiting timeout with https://github.com/OpenSurv/OpenSurv/blob/3539bb99b4ecaa7f9915e125df7bfa3c5091d9ab/surveillance/etc/monitor1.yml#L107

**Comment by SvenVD** [comment]
throwing this in https://github.com/OpenSurv/OpenSurv/issues/47

**Comment by tmushy** [comment]
I was able to get it working, but then I decided to run apt-get upgrade and upgraded my install
Now the videos dont load and I see it trying to render on the bottom of camera frame. What player is Opensurv using? I assumed it was vlc but it doesnt seem like it in this new version. I think I need to downgrade the player 

I attached a photo of what it looks like
![20250406_161954](https://github.com/user-attachments/assets/6bd5a7b4-9538-4b6d-b791-6dacfebc1c08)

**Reply by tmushy** [reply]
> I was able to get it working, but then I decided to run apt-get upgrade and upgraded my install Now the videos dont load and I see it trying to render on the bottom of camera frame. What player is Opensurv using? I assumed it was vlc but it doesnt seem like it in this new version. I think I need to downgrade the player
> 
> I attached a photo of what it looks like ![20250406_161954](https://private-user-images.githubusercontent.com/22504621/430747445-6bd5a7b4-9538-4b6d-b791-6dacfebc1c08.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwNDcxMTQsIm5iZiI6MTc0NDA0NjgxNCwicGF0aCI6Ii8yMjUwNDYyMS80MzA3NDc0NDUtNmJkNWE3YjQtOTUzOC00YjZkLWI3OTEtNmRhY2ZlYmMxYzA4LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDE3MjY1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkYWFkNmM3OWYyNmUyZmVhMjE4YTlmOTY0ZDYwMWI3YmI2ZmQyNmY1MjJiYzhiNTRlMDg1NThjNWQ3MmQyNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LTt4WTeI2w9rflDDXG5NReooHqlVR-9yo17qIfatnTg)

I gave up on the pi3 after devoting a day to it. Found my pi4 and that worked instantly. Much better than the pi3. Got 7 cameras streaming just fine

**Comment by mnrphoto** [comment]
PI4 and I'm not able to get it going. when i start the service i get the login screen only. installed python3-yaml but that did not change anything.

Getting following error in the log:

2025/05/01 13:25:26 - DAEMON:
2025/05/01 13:25:26 - DAEMON: (xfwm4:924): xfwm4-WARNING **: 13:25:26.291: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not$
2025/05/01 13:25:26 - l_default - INFO - Starting opensurv 1.3
2025/05/01 13:25:26 - DAEMON: Traceback (most recent call last):
2025/05/01 13:25:26 - DAEMON: File "surveillance.py", line 87, in <module>
2025/05/01 13:25:26 - DAEMON: monitors=get_monitors()
2025/05/01 13:25:26 - DAEMON: File "surveillance.py", line 18, in get_monitors
2025/05/01 13:25:26 - DAEMON: for m in root.xrandr_get_monitors().monitors:
2025/05/01 13:25:26 - DAEMON: AttributeError: 'Window' object has no attribute 'xrandr_get_monitors'
2025/05/01 13:25:26 - DAEMON: xfwm4: Fatal IO error 11 (Resource temporarily unavailable) on X server :0.
2025/05/01 13:25:27 - DAEMON: XIO:  fatal IO error 2 (No such file or directory) on X server ":0"
2025/05/01 13:25:27 - DAEMON: after 23 requests (23 known processed) with 0 events remaining.

**Comment by jcrossbdn** [comment]
I have spent a great deal of time with the RPI3/Bookworm configuration and have found that it may work for a little bit of time (3-4 RTSP streams) but then will just stop working with no explanation after a random period of time.  I switched the SD card to a RPI4 and it has now been running for 12 hours with no issues so far.

I would agree with others in this thread and vote that RPI3/Bookworm should perhaps not have a compatible label.

**Reply by tmushy** [reply]
I spent a full day and had the same issues. Its a no go. My pi4 on the other hand has been rock solid with 8 streams going to it

**Comment by SvenVD** [comment]
Thanks for all the efforts, based on all the comments I removed the compatible label and added the "not supported" label

**Comment by SvenVD** [comment]
Although I do not recommend it but if you really want something to work for this (old and non-generic) hardware better keep on using https://github.com/SvenVD/rpisurv (EOL) which was tailored to work on Raspberry Pi, or upgrade to a compatible OpenSurv hardware
